src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -279,7 +279,8 @@ void CodegenLLVMVisitor::visit_procedure_or_function(const ast::Block& node) {
     const auto& statements = block->get_statements();
     for (const auto& statement: statements) {
         // \todo: Support other statement types.
-        if (statement->is_local_list_statement() || statement->is_expression_statement())
+        if (statement->is_local_list_statement() || statement->is_expression_statement() ||
+            statement->is_while_statement())
```

Comment:
Just a minor suggestion that could go in a future PR if you like. I saw that the following expression is reused multiple times.
```
statement->is_local_list_statement() || statement->is_expression_statement() ||
            statement->is_while_statement()
```
Maybe it would make sense to create a new function to replace this expression and avoid rewriting it and forgetting any new type of statements while they are added? Like `is_supported(statement)`?

Reply:
Indeed! I did that in previous PR (maybe that's why submitting many PRs on a single day is not a good idea😊) with if/else support. This one have not been rebased, so yes, this `is_while_statement()` function will go into `is_supported()`.

But thanks for pointing that out! I think some refactoring is needed later as some repetitive patterns started to occur.

Reply:
Ah yes! I also forgot that, sorry :slightly_smiling_face:
Yes, this is something that can be done step by step. No worries :+1: